code/modules/spells/spell.dm

```diff
@@ -1,14 +1,44 @@
 #define TARGET_CLOSEST 1
 #define TARGET_RANDOM 2
 
+
 /obj/effect/proc_holder
 	var/panel = "Debug"//What panel the proc holder needs to go on.
+	var/active = 0 //Used by toggle based abilities.
 
 var/list/spells = typesof(/obj/effect/proc_holder/spell) //needed for the badmin verb for now
 
-/obj/effect/proc_holder/proc/InterceptClickOn(mob/user, params, atom/A)
-	return
-
+/obj/effect/proc_holder/proc/InterceptClickOn(mob/living/user, params, atom/A)
+	if(user.ranged_ability && user.ranged_ability != src)
+		user << "<span class='warning'><b>[user.ranged_ability.name]</b> has been replaced by <b>[name]</b>."
+		user.ranged_ability.remove_ranged_ability(user)
+		return 1 //1 for failed, 0 for passed.
+	user.next_click = world.time + 6
```

Comment:
I will have to make a new one, because I do not believe one exist for this exact value.